packages/gatsby/src/redux/actions.js

```diff
@@ -87,6 +97,68 @@ actions.createPage = (page, plugin = ``, traceId) => {
 }
 
 /**
+ * Delete a layout
+ * @param {string} layout a layout object with at least the name set
+ * @example
+ * deleteLayout(layout)
+ */
+actions.deleteLayout = (layout, plugin = ``) => {
+  return {
+    type: `DELETE_LAYOUT`,
+    payload: layout,
+  }
+}
+
+/**
+ * Create a layout.
+ * @param {Object} layout a layout object
+ * @param {string} page.name Unique id for layout
+ * @param {string} page.component The absolute path to the component for this layout
+ * @example
+ * createLayout({
+ *   path: `myNewLayout`,
```

Comment:
Example outdated.

Also, if you're using name as an id, perhaps we should just call it an id.